OpenChange/MAPIStoreMailFolder.m

```diff
@@ -1731,73 +1731,31 @@ - (uint32_t) exchangeRightsForRoles: (NSArray *) roles
 - (enum mapistore_error) preloadMessageBodiesWithKeys: (NSArray *) keys
                                           ofTableType: (enum mapistore_table_type) tableType
 {
+  NSEnumerator *enumerator;
+  NSUInteger max;
+  NSString *messageKey;
   MAPIStoreMailMessage *message;
-  NSMutableSet *bodyPartKeys;
-  NSMutableDictionary *keyAssoc;
-  NSDictionary *response;
-  NSUInteger count, max;
-  NSString *messageKey, *messageUid, *bodyPartKey;
-  NGImap4Client *client;
-  NSArray *fetch;
-  NSData *bodyContent;
+  NSArray* bodyContent;
 
-  if (tableType == MAPISTORE_MESSAGE_TABLE)
-    {
-      [bodyData removeAllObjects];
-      max = [keys count];
+  if (tableType != MAPISTORE_MESSAGE_TABLE)
+    return MAPISTORE_SUCCESS;
 
-      if (max > 0)
-        {
-          bodyPartKeys = [NSMutableSet setWithCapacity: max];
+  [bodyData removeAllObjects];
 
-          keyAssoc = [NSMutableDictionary dictionaryWithCapacity: max];
-          for (count = 0; count < max; count++)
-            {
-              messageKey = [keys objectAtIndex: count];
-              message = [self lookupMessage: messageKey];
-              if (message)
-                {
-                  bodyPartKey = [message bodyContentPartKey];
-                  if (bodyPartKey)
-                    {
-                      [bodyPartKeys addObject: bodyPartKey];
-                      messageUid = [self messageUIDFromMessageKey: messageKey];
-                      /* If the bodyPartKey include peek, remove it as it is not returned
-                         as key in the IMAP server response.
-
-                         IMAP conversation example:
-                         a4 UID FETCH 1 (UID BODY.PEEK[text])
-                         * 1 FETCH (UID 1 BODY[TEXT] {1677}
-                      */
-                      bodyPartKey = [bodyPartKey stringByReplacingOccurrencesOfString: @"body.peek"
-                                                                           withString: @"body"];
-                      [keyAssoc setObject: bodyPartKey forKey: messageUid];
-                    }
-                }
-            }
+  max = [keys count];
+  if (max == 0)
+    return MAPISTORE_SUCCESS;
 
-          client = [[(SOGoMailFolder *) sogoObject imap4Connection] client];
-          [client select: [sogoObject absoluteImap4Name]];
-          response = [client fetchUids: [keyAssoc allKeys]
-                             parts: [bodyPartKeys allObjects]];
-          fetch = [response objectForKey: @"fetch"];
-          max = [fetch count];
-          for (count = 0; count < max; count++)
+  enumerator = [keys objectEnumerator];
+  while ((messageKey = [enumerator nextObject]))
+    {
+      message = [self lookupMessage: messageKey];
+      if (message)
+        {
+          bodyContent = [message getBodyContent];
```

Comment:
This part of the code requires the body content is requested using `peek` feature [1] in order to avoid increasing the modseq on these messages and avoid setting the `\seen` flag.

See #50 and #105 for details.

Could you ensure that?

[1] https://tools.ietf.org/html/rfc3501 (Search for BODY.PEEK)